sonar-dotnet-shared-library/src/test/java/org/sonar/plugins/dotnet/tests/UnitTestResultsImportSensorTest.java

```diff
@@ -143,44 +102,89 @@ public void describe_execute_only_when_key_present() {
     });
     DefaultSensorDescriptor descriptor = new DefaultSensorDescriptor();
 
-    new UnitTestResultsImportSensor(unitTestResultsAggregator, ProjectDefinition.create(), "cs", "C#", analysisWarnings)
-      .describe(descriptor);
+    new UnitTestResultsImportSensor(unitTestResultsAggregator, metadata, analysisWarnings).describe(descriptor);
 
     assertThat(descriptor.configurationPredicate()).accepts(configWithKey);
     assertThat(descriptor.configurationPredicate()).rejects(configWithoutKey);
   }
 
   @Test
-  public void describe_is_global_and_only_on_language() {
+  public void describe_only_on_language() {
     UnitTestResultsAggregator unitTestResultsAggregator = mock(UnitTestResultsAggregator.class);
+    DotNetPluginMetadata metadata = mockCSharpMetadata();
     AnalysisWarnings analysisWarnings = mock(AnalysisWarnings.class);
     DefaultSensorDescriptor descriptor = new DefaultSensorDescriptor();
 
-    new UnitTestResultsImportSensor(unitTestResultsAggregator, ProjectDefinition.create(), "cs", "C#", analysisWarnings)
-      .describe(descriptor);
+    new UnitTestResultsImportSensor(unitTestResultsAggregator, metadata, analysisWarnings).describe(descriptor);
 
-    assertThat(descriptor.isGlobal()).isTrue();
     assertThat(descriptor.languages()).containsOnly("cs");
   }
 
+  @Test
+  public void isProjectSensor() {
+    assertThat(ProjectSensor.class.isAssignableFrom(UnitTestResultsImportSensor.class)).isTrue();
+  }
+
   @Test
   public void import_two_reports_for_same_project_should_not_throw() throws Exception {
-    UnitTestResultsAggregator unitTestResultsAggregator = mock(UnitTestResultsAggregator.class);
+    UnitTestResultsAggregator aggregator = mock(UnitTestResultsAggregator.class);
+    DotNetPluginMetadata cSharpMetadata = mockCSharpMetadata();
+    DotNetPluginMetadata vbNetMetadata = mock(DotNetPluginMetadata.class);
+    when(vbNetMetadata.languageKey()).thenReturn("vb");
+    when(vbNetMetadata.languageName()).thenReturn("VB.NET");
```

Comment:
We can extract a method also for mocking vb metadata. It will make this test easier to read.